apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/BFD-2572-Cleanup-Medicare-Beneficairyid-History/invoke-psql.sh

```diff
@@ -0,0 +1,329 @@
+
+CREATE TABLE PUBLIC.MEDICARE_BENEFICIARYID_HISTORY_TEMP AS
+SELECT BENE_MBI_ID,
+	BENE_ID,
+	MBI_NUM ,
+	MBI_EFCTV_BGN_DT,
+	MBI_EFCTV_END_DT,
+	LAST_UPDATED
+FROM MEDICARE_BENEFICIARYID_HISTORY;
+
+ALTER TABLE IF EXISTS public.medicare_beneficiaryid_history_temp
+    ADD CONSTRAINT medicare_beneficiaryid_history_temp_pkey PRIMARY KEY (bene_mbi_id);
+	
+CREATE INDEX IF NOT EXISTS medicare_beneficiaryid_history_temp_bene_id_idx
+    ON public.medicare_beneficiaryid_history_temp USING btree
+    (bene_id ASC NULLS LAST);
+	
+CREATE INDEX IF NOT EXISTS medicare_beneficiaryid_history_temp_mbi_num_idx
+    ON public.medicare_beneficiaryid_history_temp USING btree
+    (mbi_num ASC NULLS LAST);
+EOM
+
+# STEP_THREE_CLEANUP_SQL
+# ==============================================================
+read -r -d '' STEP_THREE_CLEANUP_SQL << EOM
+DELETE
+FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP F
+WHERE F.BENE_MBI_ID in (
+-- matches between beneficiaries and medicare_beneficiaryid_history_temp will be deleted
+--
+SELECT A.BENE_MBI_ID
+      FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP A,
+        BENEFICIARIES B
+      WHERE B.BENE_ID = A.BENE_ID
+        AND B.MBI_NUM = A.MBI_NUM
+        AND B.EFCTV_BGN_DT = A.MBI_EFCTV_BGN_DT
+        AND (B.EFCTV_END_DT IS NULL
+                  OR B.EFCTV_END_DT = A.MBI_EFCTV_END_DT)
+UNION
+-- matches between beneficiaries_history and medicare_beneficiaryid_history_temp will be deleted
+--
+SELECT A.BENE_MBI_ID
+      FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP A,
+        BENEFICIARIES_HISTORY B
+      WHERE B.BENE_ID = A.BENE_ID
+        AND B.MBI_NUM = A.MBI_NUM
+        AND B.EFCTV_BGN_DT = A.MBI_EFCTV_BGN_DT
+        AND (B.EFCTV_END_DT IS NULL
+                  OR B.EFCTV_END_DT = A.MBI_EFCTV_END_DT) );
+EOM
+
+# STEP_FOUR_CLEANUP_SQL
+# ==============================================================
+read -r -d '' STEP_FOUR_CLEANUP_SQL << EOM
+DELETE
+FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP D
+WHERE D.BENE_MBI_ID in (
+   SELECT V1.BENE_MBI_ID
+   FROM BENEFICIARIES_HISTORY C,
+   (
+      -- find records in medicare_beneficiaryid_history that match
+      -- on bene_id but do not match on mbi_num. it simply means that
+      -- the beneficiary's current mbi_num is not equal to any of
+      -- the mbi_num(s) in the medicare_beneficiaryid_history table.
+      -- This will potentially be an 'add' if there is no matching data
+      -- already in the beneficiaries_history table.
+      --
+      SELECT A.BENE_MBI_ID,
+             A.BENE_ID,
+             A.MBI_NUM,
+             A.MBI_EFCTV_BGN_DT AS EFCTV_BGN_DT,
+             A.MBI_EFCTV_END_DT AS EFCTV_END_DT
+      FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP A,
+           BENEFICIARIES B
+      WHERE B.BENE_ID = A.BENE_ID
+      AND B.MBI_NUM <> A.MBI_NUM
+   ) V1
+-- find medicare_beneficiaryid_history_temp records that
+-- match beneficiaries_history data already present.
+WHERE (C.BENE_ID = V1.BENE_ID
+AND C.MBI_NUM = V1.MBI_NUM));
```

Comment:
This step broke my brain for a while, but I eventually got it. Filters out items that both have a bene id in beneficiaries, and then also already exist in bene history (so it'd be pointless to add the historical record since it exists in bene_history already).